server/appservices/service.go

```diff
@@ -21,22 +21,22 @@ var ErrIsABot = errors.New("is a bot")
 type Service interface {
 	// Subscriptions
 
-	Subscribe(r *incoming.Request, sub apps.Subscription) error
-	GetSubscriptions(r *incoming.Request, appID apps.AppID, actingUserID string) ([]apps.Subscription, error)
-	Unsubscribe(r *incoming.Request, sub apps.Subscription) error
+	Subscribe(*incoming.Request, apps.Subscription) error
+	GetSubscriptions(_ *incoming.Request, _ apps.AppID, actingUserID string) ([]apps.Subscription, error)
+	Unsubscribe(*incoming.Request, apps.Subscription) error
```

Comment:
@levb , what is the rule for passing `_` (instead of `r`) in the signature and when do we skip it?

Reply:
Generally in the interface declaration the argument names are not required; but for the non-intention-revealing type names like `string` they are used effectively as comments. However if one arg is named, all of them must be named. Conveniently, `_` allows skipping it.

Generally, I tried to reduce the use of names in the interface definitions, where the type itself is self-explanatory. (Since names get changed over time, stylistic ballast)